### PR TITLE
update for neptune 1.0

### PR DESCRIPTION
**Pull request by kshitij12345**
## What does this PR do?

Updates to NeptuneLogger aligned with Neptune 1.0 release.
Ref: https://docs.neptune.ai/setup/neptune-client_1-0_release_changes/

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: src/lightning/pytorch/loggers/neptune.py [hunk 1]
```diff
@@ -34,10 +34,10 @@
 from lightning.pytorch.utilities.model_summary import ModelSummary
 from lightning.pytorch.utilities.rank_zero import rank_zero_only
 
-_NEPTUNE_AVAILABLE = RequirementCache("neptune-client")
+_NEPTUNE_AVAILABLE = RequirementCache("neptune") or RequirementCache("neptune-client")
 if _NEPTUNE_AVAILABLE:
-    from neptune import new as neptune
```
Comment (kshitij12345) [inline src/lightning/pytorch/loggers/neptune.py:39]: Figure out the way to import from `neptune` depending on pre and post 1.0.

**Comment by twolodzko** [comment]
You missed one `.log` in `log_metrics` line 424

**Comment by twolodzko** [comment]
* In the docstrings, you have the deprecated "ANONYMOUS" instead of `neptune.ANONYMOUS_API_TOKEN`.
* I might have missed it, but it'd be good to have unit tests on using the handlers instead of Run.

File: docs/source-pytorch/visualize/supported_exp_managers.rst [hunk 2]
```diff
@@ -68,23 +68,24 @@ To use `Neptune.ai <https://neptune.ai/>`_ first install the neptune package:
...
 
-    pip install neptune-client
+    pip install neptune
 
 or with conda:
 
 .. code-block:: bash
 
-    conda install -c conda-forge neptune-client
+    conda install -c conda-forge neptune
```
Comment (kshitij12345) [inline docs/source-pytorch/visualize/supported_exp_managers.rst:77]: Conda is still being updated.

File: src/lightning/pytorch/loggers/neptune.py [hunk 3]
```diff
@@ -34,10 +34,10 @@
 from lightning.pytorch.utilities.model_summary import ModelSummary
 from lightning.pytorch.utilities.rank_zero import rank_zero_only
 
-_NEPTUNE_AVAILABLE = RequirementCache("neptune-client")
+_NEPTUNE_AVAILABLE = RequirementCache("neptune") or RequirementCache("neptune-client")
```
Comment (kshitij12345) [inline src/lightning/pytorch/loggers/neptune.py:37]: For compatibility with 1.0 and pre-1.0

**Comment by kshitij12345** [comment]
@twolodzko 
> In the docstrings, you have the deprecated "ANONYMOUS" instead of neptune.ANONYMOUS_API_TOKEN.

Updated!

> I might have missed it, but it'd be good to have unit tests on using the handlers instead of Run.

I am not sure if we can easily mock the handlers. Though I have tested locally and made a few changes as `wait` and `exists` method are not supported on `Handler`.

PTAL :)

File: requirements/pytorch/loggers.info [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # all supported loggers. this list is here as a reference, but they are not installed in CI
-neptune>=1.0.0
+neptune
```
Comment (twolodzko) [inline requirements/pytorch/loggers.info:2]: >= 0.16.16 for it to support `.append` and `stringify_unsupported`

File: src/lightning/pytorch/loggers/neptune.py (hunk 3 above)
Comment (twolodzko) [reply src/lightning/pytorch/loggers/neptune.py:37]: Same as above.

**Review by twolodzko — APPROVED** [review]

File: requirements/pytorch/loggers.info (hunk 4 above)
Comment (kshitij12345) [reply requirements/pytorch/loggers.info:2]: I think this should be non-issue as `neptune` package is available only from 1.0 onwards, so both should be available.

Ref: https://pypi.org/project/neptune/#history

**Review by AleksanderWWW — APPROVED** [review]

**Comment by kshitij12345** [comment]
cc: @Blaizzy

**Comment by kshitij12345** [comment]
cc: @normandy7 for reviewing documentation once :)

**Review by Blaizzy — APPROVED** [review]
LGTM!

**Review by Borda — APPROVED** [review]

File: src/lightning/pytorch/loggers/neptune.py (hunk 3 above)
Comment (awaelchli) [reply src/lightning/pytorch/loggers/neptune.py:37]: You can define
```py
_NEPTUNE_LT_1_0 =  RequirementCache("neptune<1.0")
```
if that helps with your logic to make it compatible both ways.

**Review by awaelchli — APPROVED** [review]
LGTM in general. 
If you could also update the changelog about the version upgrade, that would be great.

File: src/lightning/pytorch/loggers/neptune.py (hunk 3 above)
Comment (kshitij12345) [reply src/lightning/pytorch/loggers/neptune.py:37]: I don't think that would be possible as we have new package name from >1.0 onward `neptune` and `neptune-client` corresponds to packages pre 1.0 and post 1.0 for BC compatibility of requirements file.

**Comment by kshitij12345** [comment]
I see that a few CI jobs are failing due to the PR will fix them and also update the CHANGELOG. Thanks!

**Review by carmocca — COMMENTED** [review]
Please add a CHANGELOG entry

Also update these two instances:

- https://github.com/Lightning-AI/lightning/pull/16888/files#diff-66f006dbadabdd4826bf8fb6dcb776fe4b73871f73daf05f205b6e27aae7a548R1387
- https://github.com/Lightning-AI/lightning/pull/16888/files#diff-7eaf1173289ce8b1821be324737ab2a3879a7156ffb76c785adf4ecbef7cd6d9R96

**Review by normandy7 — APPROVED** [review]
Looks good otherwise.

**Comment by kshitij12345** [comment]
@Borda there are another couple of failing tests. Will fix them.

**Comment by kshitij12345** [comment]
@Borda @carmocca two failures look irrelevant (timeouts).

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Pull request merged**